src/libraries/System.Memory.Data/src/System.Memory.Data.csproj

```diff
@@ -23,7 +23,7 @@ System.BinaryData</PackageDescription>
     <Compile Include="$(CoreLibSharedDir)System\Diagnostics\CodeAnalysis\RequiresDynamicCodeAttribute.cs" />
   </ItemGroup>
 
-  <ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework)' != '$(NetCoreAppCurrent)'">
```

Comment:
This question applies for multiple files in this PR:

If a project reference is now being conditioned, why are we not also adding a new item group with reference to the package like this?:

```xml
  <ItemGroup Condition="'$(TargetFramework)' == '$(NetCoreAppCurrent)'">
    <Reference Include="System.Text.Json" />
  </ItemGroup?
```

I see you already did this in some of the files like System.Net.Http.Json.csproj and System.Text.Json.csproj.

Reply:
I also ask because the warning message says:

> Project reference '%(ProjectReference.Filename)' is a reference to a framework assembly and is not required.

Reply:
Because these are OOBs above the shared framework, so they get all the framework references by default.